Sources/StreamChat/StateLayer/ChatState+Observer.swift

```diff
@@ -55,25 +55,28 @@ extension ChatState {
         }
         
         struct Handlers {
-            let channelDidChange: (ChatChannel) async -> Void
+            let channelDidChange: (ChatChannel?) async -> Void
             let membersDidChange: (StreamCollection<ChatChannelMember>) async -> Void
             let messagesDidChange: (StreamCollection<ChatMessage>) async -> Void
             let typingUsersDidChange: (Set<ChatUser>) async -> Void
             let watchersDidChange: (StreamCollection<ChatUser>) async -> Void
         }
         
         func start(with handlers: Handlers) {
-            channelObserver.onChange(do: { change in Task { await handlers.channelDidChange(change.item) } })
-            channelObserver.onFieldChange(\.currentlyTypingUsers, do: { change in Task { await handlers.typingUsersDidChange(change.item) } })
-            memberListObserver = memberListState.$members.sink(receiveValue: { change in Task { await handlers.membersDidChange(change) } })
-            watchersObserver.onDidChange = { [weak watchersObserver] _ in
-                guard let items = watchersObserver?.items else { return }
-                let collection = StreamCollection(items)
-                Task { await handlers.watchersDidChange(collection) }
-            }
+            memberListObserver = memberListState.$members
+                .dropFirst() // skip initial
+                .sink(receiveValue: { change in Task { await handlers.membersDidChange(change) } })
             
             do {
-                try channelObserver.startObserving()
+                var lastTypingUsers: Set<ChatUser>?
+                try channelObserver.startObserving(didChange: { channel in
+                    let currentlyTypingUsers: Set<ChatUser> = channel?.currentlyTypingUsers ?? Set()
+                    if lastTypingUsers != currentlyTypingUsers {
+                        lastTypingUsers = currentlyTypingUsers
+                        await handlers.typingUsersDidChange(currentlyTypingUsers)
+                    }
+                    await handlers.channelDidChange(channel)
+                })
```

Comment:
Similar to [onFieldChange](https://github.com/GetStream/stream-chat-swift/blob/d2e1e594ca8084d36a772d6632e86d7050574add/Sources/StreamChat/Controllers/DatabaseObserver/EntityDatabaseObserver.swift#L273) but I was not sure if it is worth it to add a similar function to the new observer. Open for suggestions.

Reply:
I think it makes asense to remove it indeed

Reply:
Yeah, I think it makes a lot of sense as well.

Reply:
yep, totally agree with removing it.